arangod/VocBase/vocbase.cpp

```diff
@@ -1014,25 +1014,25 @@ std::string TRI_vocbase_t::viewName(TRI_voc_cid_t id) const {
 }
 
 /// @brief looks up a collection by name or stringified cid or uuid
-arangodb::LogicalCollection* TRI_vocbase_t::lookupCollection(
+std::shared_ptr<arangodb::LogicalCollection> TRI_vocbase_t::lookupCollection(
     std::string const& nameOrId
 ) const noexcept {
   return std::dynamic_pointer_cast<arangodb::LogicalCollection>(
     lookupDataSource(nameOrId)
-  ).get();
```

Comment:
use the category instead of dynamic_cast ?

Reply:
done